### PR TITLE
Barcode scan

### PR DESCRIPTION
**Pull request by s0rta**
added a minimal barcode scan implementation using the image labeling implementation as a framework. I've got it up on npm under barcode-scanning right now (username notyet) but wanted commit here as you have a great organizational structure to the mlkit implementations. Right now it just scans all barcodes, and returns an array of objects containing the value and the type of barcode. should be working for android and iOS. If there are any changes needed I'm down to work on them, just let me know.

### CODE REVIEW

File: barcode-scanning/package.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
+{
+  "name": "barcode-scanning",
```
Comment (a7med-mahmoud) [inline barcode-scanning/package.json:2]: This should be `@react-native-ml-kit/barcode-scanning`.

File: barcode-scanning/android/src/main/java/com/rnmlkit/barcodescanning/BarcodeScanningModule.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        return "BarcodeScanning";
+    }
+
+    @ReactMethod
+    public void scan(String url, Promise promise) {
+        Uri uri = Uri.parse(url);
+        InputImage image;
+        try {
+            image = InputImage.fromFilePath(reactContext, uri);
+            // BarcodeScannerOptions options =
```
Comment (a7med-mahmoud) [inline barcode-scanning/android/src/main/java/com/rnmlkit/barcodescanning/BarcodeScanningModule.java:47]: You can remove these lines if we don't need them.

**Review by a7med-mahmoud — CHANGES_REQUESTED** [review]
Thanks a lot, @s0rta!
I think it would be better if you changed `readme.md` to `README.md` just for consistency.
Please resolve the little hints I added. I will test it and let you know if there's anything else need.

File: barcode-scanning/android/src/main/java/com/rnmlkit/barcodescanning/BarcodeScanningModule.java (hunk 2 above)
Comment (s0rta) [reply barcode-scanning/android/src/main/java/com/rnmlkit/barcodescanning/BarcodeScanningModule.java]: removed the comment but confused about removing the image input, doesn't the scanner need an image to process?

Comment (a7med-mahmoud) [reply barcode-scanning/android/src/main/java/com/rnmlkit/barcodescanning/BarcodeScanningModule.java]: I was talking about the comment, I should've been more clear. Thanks!

**Comment by a7med-mahmoud** [comment]
Can you add an example app to it? To make it easier to test the package you added. You can copy an example app from any of the other modules in the repo and apply the changes needed.

**Review by a7med-mahmoud — APPROVED** [review]
I've added the required changes. Now we have basic support for barcode scanning 🎉. Thanks for the contribution, @s0rta!

**Pull request merged**